src/DataAPI.jl

```diff
@@ -515,6 +515,97 @@ Throw an error if `x` does not support metadata deletion for column `col`.
 """
 function emptycolmetadata! end
 
+"""
+    dimmetadatasupport(T::Type, dim::Int)
+
+Return a `NamedTuple{(:read, :write), Tuple{Bool, Bool}}` indicating whether
+values of type `T` support metadata correspond to dimension `dim`.
```

Comment:
Sorry, I missed this substantial point: `colmetadatasupport` takes a single argument, i.e. it asks whether column metadata is supported at all, without considering a particular column. Shouldn't we do the same here? Once a type supports dimension metadata, it can return `nothing` for dimensions that do not have any metadata.

Reply:
What do you mean by "there is exactly one dimension always"? :-)